armor-write/src/main/java/com/rapid7/armor/write/writers/ColumnFileWriter.java

```diff
@@ -307,12 +309,354 @@ private void load(DataInputStream inputStream) throws IOException {
           .filter(e -> e.getNumRows() > 0).collect(Collectors.toList());
     }
   }
-  
+
   public StreamProduct buildInputStream(Compression compress) throws IOException {
+    return buildInputStream(compress, DEFAULT_VERSION);
+  }
+
+  public StreamProduct buildInputStream(Compression compress, Constants.ColumnFileFormatVersion version) throws IOException {
+    switch (version) {
+      case VERSION_1:
+        return buildInputStreamV1(compress);
+      case VERSION_2:
+        return buildInputStreamV2(compress);
+      default:
+        throw new IllegalArgumentException("Unknown column file format version " + version);
+    }
+  }
+
+  static class SectionBuilder {
+    private final ColumnFileSection sectionType;
+    private ByteArrayOutputStream outputStream;
+
+    public ByteArrayOutputStream getOutputStream() {
+      return this.outputStream;
+    }
+
+    public SectionBuilder(ColumnFileSection header)
+    {
+      this.sectionType = header;
+      this.outputStream = new ByteArrayOutputStream();
+    }
+
+    public Section build() {
+      return new SingleSection(this.sectionType, this.outputStream.toByteArray());
+    }
+  }
+
+  static interface SectionBlob {
+    int getLength();
+    InputStream getInputStream();
+  }
+
+  static class SingleSectionBlob implements SectionBlob {
```

Comment:
I think we could probably rename this to like ByteArraySectionBlob

Reply:
fixed.